Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -220,6 +220,10 @@ void GcodeSuite::G28(const bool always_home_all) {
                homeZ = always_home_all || parser.seen('Z'),
                home_all = (!homeX && !homeY && !homeZ) || (homeX && homeY && homeZ);
 
+    #if ENABLED(BLTOUCH)
+      if (!HOMING_Z_WITH_PROBE || home_all || homeX || homeY) set_bltouch_deployed(false);
```

Comment:
You can't use compiler directives like `defined`, `ENABLED`, etc. in an `if` statement. So `HOMING_Z_WITH_PROBE` won't work here. You can only use it in `#if`.